crossbeam-channel/src/flavors/array.rs

```diff
@@ -542,7 +542,17 @@ impl<T> Drop for Channel<T> {
             };
 
             unsafe {
-                self.buffer.add(index).drop_in_place();
+                let ptr = self.buffer.add(index);
+                {
+                    // This requires an extra scope because when we drop the Slot,
+                    // reference to it should not exist.
+                    let slot = &mut *ptr;
+                    let msg = &mut *slot.msg.get();
+                    // Drop the message (MaybeUninit<T>).
+                    msg.as_mut_ptr().drop_in_place();
+                }
+                // Drop slot (This should be a no-op).
+                ptr.drop_in_place();
```

Comment:
Should we leave this in or remove it, since it's clearly a no-op?

Reply:
I'm fine either way, but we could clean this up quite a bit by removing `ptr.drop_in_place()` because then we can also remove the extra scope, right?

Reply:
Sounds good, will push the remaining changes very soon.